File: baekjoon/src/tags/dynamic/No1003.java
package tags.dynamic;

import java.io.*;
import java.util.StringTokenizer;

/**
 * https://www.acmicpc.net/problem/1003
 */
public class No1003 {
    public static final BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));

    public static final BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));

    public static final int[] fibonacci = new int[41];

    public static void calculateFibonacci() {
        fibonacci[0] = 0;
        fibonacci[1] = 1;
        for (int i = 2; i < fibonacci.length; i++)
            fibonacci[i] = fibonacci[i - 2] + fibonacci[i - 1];
    }

    public static void main(String[] args) throws IOException {
        calculateFibonacci();

        int testCase = Integer.parseInt(reader.readLine());
        for (int i = 0; i < testCase; i++) {
            int number = Integer.parseInt(new StringTokenizer(reader.readLine()).nextToken());
            if (number == 0)
                writer.write("1 0");
            else
                writer.write(fibonacci[number - 1] + " " + fibonacci[number]);
            writer.newLine();
            writer.flush();
        }
    }
}

File: baekjoon/src/level/silver3/No1003.kt
package level.silver3

import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

/**
 * https://www.acmicpc.net/problem/1003
 */
private val reader = BufferedReader(InputStreamReader(System.`in`))
private val writer = BufferedWriter(OutputStreamWriter(System.out))

private lateinit var fibonacci: Array<IntArray>

fun main() {
    setFibonacci()

    val testCase = reader.readLine().toInt()
    repeat(testCase) {
        val number = reader.readLine().toInt()
        writer.write("${fibonacci[0][number]} ${fibonacci[1][number]}\n")
    }
    writer.flush()
}

private fun setFibonacci() {
    fibonacci = Array(2) { IntArray(41) }
    fibonacci[0][0] = 1
    fibonacci[1][1] = 1
    for (i in 2..40) {
        fibonacci[0][i] = fibonacci[0][i - 1] + fibonacci[0][i - 2]
        fibonacci[1][i] = fibonacci[1][i - 1] + fibonacci[1][i - 2]
    }
}